packages/react-ui-components/src/components/CodeEditor/types.ts

```diff
@@ -1,8 +1,29 @@
 import type { editor } from 'monaco-editor';
+import { FunctionComponent } from 'react';
 
 export type OnEditorChangeCallback = (value: string | undefined) => void;
 export type CustomThemeData = editor.IStandaloneThemeData;
-export type CustomEditorOptions = editor.IEditorOptions & editor.IGlobalEditorOptions;
+
+export type KvEditorPaddingOptions = {
+	top?: number;
+	bottom?: number;
+};
+
+export type KvEditorOptions = {
```

Comment:
we are creating a limitation on editor options to only allow passing these properties. 
We should extend all editor options here. In rules/process map editor we are using directly the monaco-editor, and if we refactor these components to use this component we should allow the other monaco options here, because they use other options.

Reply:
I'm torn between these 2 approaches, exposing everything monaco has vs having this intermediate type that allows us to have some abstraction over what editor we use underneath.

@djcouto what's your take?

Reply:
IMO, I'm a fan of keeping this library agnostic. Today we are using monaco but it can run unmaintained tomorrow or a new cutting-edge library can come up. With that being said, these settings are a bit monaco-specific, so I would say to go with @miguelpinto98's suggestion and expose everything monaco has.